processing/src/main/java/org/apache/druid/query/DataSource.java

```diff
@@ -35,7 +35,8 @@
     @JsonSubTypes.Type(value = UnionDataSource.class, name = "union"),
     @JsonSubTypes.Type(value = JoinDataSource.class, name = "join"),
     @JsonSubTypes.Type(value = LookupDataSource.class, name = "lookup"),
-    @JsonSubTypes.Type(value = InlineDataSource.class, name = "inline")
+    @JsonSubTypes.Type(value = InlineDataSource.class, name = "inline"),
+    @JsonSubTypes.Type(value = GlobalTableDataSource.class, name = "global")
```

Comment:
Should the `global` datasource type be added to the docs? https://druid.apache.org/docs/0.18.1/querying/query-execution.html#datasource-type

Reply:
I'm not sure if it makes sense to add yet until more of #9953 is implemented since making effective use of this requires `GlobalTableDataSource` to be bound to a `JoinableFactory`

Reply:
By the way, to be clear, I don't think we need to do (2) in this PR. That was more a suggestion for future work.

Reply:
Renamed to `globalTable`. For 2 it sounds nice to be automatic since it negates the need to document this, so will add that behavior in a follow-up.